xml/hwbp_registers.xml

```diff
@@ -95,11 +95,23 @@
         <field name="data" bits="XLEN-1:0" access="R/W" reset="Preset" />
     </register>
 
-    <register name="Trigger Data 3" short="tdata3" address="0x7a3">
-        Trigger-specific data.
+    <register name="Trigger Extra" short="textra" address="0x7a3">
+        \begin{commentary}
+            ASIDLEN is limited in size when XLEN is less than DXLEN to preserve
+            the maximum number of bits for future use.
+        \end{commentary}
+        <field name="0" bits="XLEN-1:2+ASIDLEN" access="R" reset="0" />
+        <field name="tasid" bits="1+ASIDLEN:1" access="R/W" reset="0">
+            If XLEN is 32 then ASIDLEN is 9. Otherwise ASIDLEN is 16.
+        </field>
+        <field name="tasiden" bits="0" access="R/W" reset="0">
+            0: \Ftasid is ignored.
 
-        If XLEN is less than DXLEN, writes to this register are sign-extended.
-        <field name="data" bits="XLEN-1:0" access="R/W" reset="Preset" />
+            1: This trigger will only match if \Fasid in \Rsatp equals the
+            lower bits of \Ftasid. The number of bits to compare depends on the
+            current value of SXLEN. If it is 32, then 9 bits are compared,
```

Comment:
Because ASID is in the satp register, which the privspec says is SXLEN bits wide. It then makes ASID dependent on SXLEN exactly like I did here.

Reply:
Yeah, I didn't realize that `\Fasid` was already a thing, I thought we had to have it added to the spec.

Then I guess the question (as discussed in the email thread) is whether `asid` is acceptable standin for a "process" or whether we need a specific CSR for this. My gut is we should go with the `asid` already defined in the spec, but I'm not an OS programmer

Reply:
I imagine operating systems would use `asid` to prevent processes from walking on each other's memory, so in UNIX systems every process would have its own `asid`. Threads within processes might share `asid` though, and I'm sure there are other systems where a context ID would be better. I'd rather not define a context ID in the debug spec, though. If we do, it should be in a separate PR after this one merges.